packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -412,19 +412,19 @@ export default class RouterService extends Service {
     half-filled out:
 
     ```app/routes/contact-form.js
-    import Route from '@ember/routing';
+    import Route from '@ember/routing/route';
     import { action } from '@ember/object';
     import { inject as service } from '@ember/service';
 
-    export default class extends Route {
+    export default class ContactFormRoute extends Route {
       @service router;
 
       constructor() {
         super(...arguments);
 
         this.router.on('routeWillChange', (transition) => {
           if (!transition.to.find(route => route.name === this.routeName)) {
-            alert("Please save or cancel your changes.");
+            alert('Please save or cancel your changes.');
```

Comment:
Should this be a `confirm` in the if expression?

Reply:
confirm seems right!